src/TestableIO.System.IO.Abstractions/IFileSystemExtensionPoint.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace System.IO.Abstractions
+{
+    /// <summary>
+    ///     Interface to support implementing extension methods on top of nested <see cref="IFileSystem" /> interfaces.
+    /// </summary>
+    public interface IFileSystemExtensionPoint
```

Comment:
Can we maybe give this name a bit more meaning?

```suggestion
    public interface IFileSystemEntity
```